Formula/o/openjdk@8.rb

```diff
@@ -126,6 +137,14 @@ def install
           --with-extra-cxxflags=-F#{javavm_framework_path}
         ]
         ldflags << "-F#{javavm_framework_path}"
+        # Fix "'JavaNativeFoundation/JavaNativeFoundation.h' file not found" issue on MacOS Sonoma.
+      elsif MacOS.version == :sonoma
+        javavm_framework_path = "/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks"
+        args += %W[
+          --with-extra-cflags=-F#{javavm_framework_path}
+          --with-extra-cxxflags=-F#{javavm_framework_path}
+        ]
+        ldflags << "-F#{javavm_framework_path}"
```

Comment:
Does this work:

```suggestion
      elsif MacOS.version == :sonoma
        # Fix "'JavaNativeFoundation/JavaNativeFoundation.h' file not found" issue on MacOS Sonoma.
        sdk_path = MacOS::CLT.sdk_path(MacOS.version)
        ENV["SDKPATH"] = ENV["SDKROOT"] = sdk_path
        frameworks_path = sdk_path/"System/Library/Frameworks"
        args += %W[
          --with-extra-cflags=-F#{frameworks_path}
          --with-extra-cxxflags=-F#{frameworks_path}
        ]
        ldflags << "-F#{frameworks_path}"
```

It doesn't look right to use the macOS 13 SDK on macOS 14 (Sonoma).

Reply:
> Does this work? ......

I tested locally, this cannot be compiled successfully:
```
Making signal interposition lib...
Making SA debugger back-end...
/private/tmp/openjdkA8-20240306-9300-slp1a1/jdk8u-jdk8u402-ga/hotspot/agent/src/os/bsd/MacosxDebuggerLocal.m:27:9: fatal error: 'JavaNativeFoundation/JavaNativeFoundation.h' file not found
#import <JavaNativeFoundation/JavaNativeFoundation.h>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/private/tmp/openjdkA8-20240306-9300-slp1a1/jdk8u-jdk8u402-ga/hotspot/agent/src/os/bsd/MacosxDebuggerLocal.m:27:9: note: did not find header 'JavaNativeFoundation.h' in framework 'JavaNativeFoundation' (loaded from '/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks')
1 error generated.
make[6]: *** [libsaproc.dylib] Error 1
make[6]: *** Waiting for unfinished jobs....
make[5]: *** [the_vm] Error 2
make[4]: *** [product] Error 2
make[3]: *** [generic_build2] Error 2
make[2]: *** [product] Error 2
make[1]: *** [/private/tmp/openjdkA8-20240306-9300-slp1a1/jdk8u-jdk8u402-ga/build/release/hotspot/_hotspot.timestamp] Error 2
make: *** [hotspot-only] Error 2

==> Formula
Path: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/o/openjdk@8.rb
==> Configuration
HOMEBREW_VERSION: 4.2.11-36-gd55fa09
ORIGIN: https://github.com/Homebrew/brew
HEAD: d55fa09d1ee877859dd22f4f8129f7fa024dbb72
Last commit: 8 hours ago
Core tap origin: https://github.com/<***>/homebrew-core.git
Core tap HEAD: 4d8f0739f62b1ff66dbfe3cdfb73759335583d46
Core tap last commit: 22 hours ago
Core tap branch: openjdk@8u402
Core tap JSON: 06 Mar 02:53 UTC
Core cask tap JSON: 06 Mar 02:53 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_NO_INSTALL_FROM_API: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.44.0 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.3.1-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.2
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
SDKROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk
MAKEFLAGS: JOBS=12
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_PATH: /usr/local/opt/libpng/lib/pkgconfig:/usr/local/opt/freetype/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/14
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/local/opt/autoconf/bin:/usr/local/opt/pkg-config/bin:/usr/local/opt/libpng/bin:/usr/local/opt/freetype/bin:/usr/local/opt/giflib/bin:/usr/local/opt/gawk/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: openjdk@8 1.8.0-402 did not build
Logs:
     /Users/<***>/Library/Logs/Homebrew/openjdk@8/00.options.out
     /Users/<***>/Library/Logs/Homebrew/openjdk@8/01.bash
     /Users/<***>/Library/Logs/Homebrew/openjdk@8/02.bash
     /Users/<***>/Library/Logs/Homebrew/openjdk@8/02.bash.cc
     /Users/<***>/Library/Logs/Homebrew/openjdk@8/03.make
     /Users/<***>/Library/Logs/Homebrew/openjdk@8/03.make.cc
     /Users/<***>/Library/Logs/Homebrew/openjdk@8/build
READ THIS: https://docs.brew.sh/Troubleshooting
```

> It doesn't look right to use the macOS 13 SDK on macOS 14 (Sonoma).

That's no way to do this, dependent `JavaNativeFoundation.Framework` in `MacOSX14.sdk`  without `Headers` and `Modules`, But they do exist in `MacOSX13.sdk` (not sure if it's a MacOS Sonoma Bug). My intention is that only this `JavaNativeFoundation.Framework` uses the `MacOSX13.sdk`'s `Headers`(maybe and/or `Modules`), and the others still use the `MacOSX14.sdk`'s, so I did not change the `SDKPATH` and `SDKROOT` ENV values on MacOS Sonoma.

```
➜  ~ ✗ ll /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks/JavaNativeFoundation.framework/Versions/A
total 8
drwxr-xr-x  19 root  wheel   608B  3  6 10:01 Headers
-rw-r--r--   1 root  wheel   3.8K  2 18  2023 JavaNativeFoundation.tbd
drwxr-xr-x   3 root  wheel    96B  3  6 10:01 Modules
➜  ~ ✗ ll /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/System/Library/Frameworks/JavaNativeFoundation.framework/Versions/A
total 8
-rw-r--r--  1 root  wheel   3.8K  2 10 18:03 JavaNativeFoundation.tbd
```

Reply:
> It's intentional because it's removed from macOS 14 SDK unfortunately. We will have a problem come macOS 15 where there is no macOS 13 SDK to use but that's a problem for a few months time - hopefully upstream has a better solution by then.

I don't know if this is intentional by Apple or if it's a bug in Sonoma.

Reply:
It's likely intentional.